### PR TITLE
add superuser fallback to authorizer

### LINKED ISSUE #109211: admission chain is given authorizer that doesn't include superuser fallback
### What happened?

I created a v1.21.5 cluster. After I deleted the default cluster-admin clusterrolebinding with group "system: master", I failed to approve a pending csr with "kubectl certificate approve csr" and get the following error
Error from server (Forbidden): certificatesigningrequests.certificates.k8s.io "csr-abcdefg" is forbidden: user not permitted to approve requests with signerName "kubernetes.io/kubelet-serving"

### What did you expect to happen?

the clusterrolebinding is not needed for the system:masters group to have superuser permissions, delete the clusterrolebinding shouldn't cause loss of permissions for the system:masters group

### How can we reproduce it (as minimally and precisely as possible)?

- Create a cluster
- kubectl delete clusterrolebinding cluster-admin
- Create a csr in the cluster
- Try to approve that csr with `kubectl certificate approve`

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
v1.21.5
```

</details>

### Cloud provider

<details>
vmware
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by mzkhan**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:
Fixes #109211

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubeapiserver/authorizer/config.go [hunk 1]
```diff
@@ -144,5 +145,9 @@ func (config Config) New() (authorizer.Authorizer, authorizer.RuleResolver, erro
 		}
 	}
 
+	// Add SystemPrivilegedGroup as an authorizing group
```
[the remark this answers is not in the document]
Comment (liggitt) [reply pkg/kubeapiserver/authorizer/config.go:148]: we should also be able to drop it from https://github.com/kubernetes/kubernetes/blob/75e49ec824b183288e1dbaccfd7dbe77d89db381/staging/src/k8s.io/apiserver/pkg/server/config.go#L939-L940

and would need a test demonstrating this authorizer is effective for admission

**Comment by liggitt** [comment]
master is open for 1.26, would like to get this resolved early in the cycle (and possibly backport if the fix is minimal enough). Will you be able to address the comments soon?

**Comment by mzkhan** [comment]
> Will you be able to address the comments soon?

Sure. I should be able to.

**Comment by liggitt** [comment]
the integration test failures look related... I suspect the failing tests are using a different method of setting up the test server which does not go through the normal authorizer setup path

**Comment by liggitt** [comment]
yeah, it looks like some integration tests are monkeying with the authorizer:

```
 git grep ".Authorization.Authorizer = " -- test/integration
test/integration/apiserver/flowcontrol/concurrency_util_test.go:                        config.GenericConfig.Authorization.Authorizer = authz
test/integration/auth/accessreview_test.go:                     config.GenericConfig.Authorization.Authorizer = sarAuthorizer{}
test/integration/auth/accessreview_test.go:                     config.GenericConfig.Authorization.Authorizer = sarAuthorizer{}
test/integration/auth/accessreview_test.go:                     config.GenericConfig.Authorization.Authorizer = sarAuthorizer{}
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysDenyAuthorizer()
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = allowAliceAuthorizer{}
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = allowAliceAuthorizer{}
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = allowAliceAuthorizer{}
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = impersonateAuthorizer{}
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = trackingAuthorizer
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = a
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = a
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = a
test/integration/auth/auth_test.go:                     config.GenericConfig.Authorization.Authorizer = allowAliceAuthorizer{}
test/integration/auth/bootstraptoken_test.go:                                   config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/auth/dynamic_client_test.go:                   config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/auth/selfsubjectreview_test.go:                        config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/auth/selfsubjectreview_test.go:                        config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/auth/svcaccttoken_test.go:                     config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
test/integration/controlplane/synthetic_controlplane_test.go:   config.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysDenyAuthorizer()
test/integration/controlplane/synthetic_controlplane_test.go:   config.GenericConfig.Authorization.Authorizer = allowAliceAuthorizer{}
test/integration/serviceaccount/service_account_test.go:                        config.GenericConfig.Authorization.Authorizer = authorizer
```

Previously, whatever the integration test set had the superuser authorizer prepended... now, the integration test is stomping the superuser authorizer.

**Comment by mzkhan** [comment]
> Previously, whatever the integration test set had the superuser authorizer prepended... now, the integration test is stomping the superuser authorizer.

I checked by adding back the superuser authorizer to the file `staging/src/k8s.io/apiserver/pkg/server/config.go`, and the tests starts succeeding. Looks like those tests depend on the authorizer being added that way.

**Comment by liggitt** [comment]
> > Previously, whatever the integration test set had the superuser authorizer prepended... now, the integration test is stomping the superuser authorizer.
> 
> I checked by adding back the superuser authorizer to the file `staging/src/k8s.io/apiserver/pkg/server/config.go`, and the tests starts succeeding. Looks like those tests depend on the authorizer being added that wa

I opened https://github.com/kubernetes/kubernetes/pull/112590 with three commits making the authn/authz customizations in those test packages coexist with default API server authn/authz. Once I get those commits green, we can merge those test changes separately

**Comment by mzkhan** [comment]
> I think we'd rather fix those tests than leave the authz stuff split in two places.

Sounds good.

File: pkg/kubeapiserver/authorizer/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (liggitt) [reply pkg/kubeapiserver/authorizer/config.go]: still needs the test demonstrating this is effective for admission

[the remark this answers is not in the document]
Comment (liggitt) [reply pkg/kubeapiserver/authorizer/config.go]: this test fails on master and passes on this PR:
```diff
diff --git a/test/integration/certificates/admission_sign_test.go b/test/integration/certificates/admission_sign_test.go
index 9e1ae99922b..de8f9d61aab 100644
--- a/test/integration/certificates/admission_sign_test.go
+++ b/test/integration/certificates/admission_sign_test.go
@@ -60,6 +60,11 @@ func TestCSRSignerNameSigningPlugin(t *testing.T) {
 			defer s.TearDownFn()
 			client := clientset.NewForConfigOrDie(s.ClientConfig)
 
+			// Drop the default RBAC superuser permissions to rely on the internal superuser authorizer
+			if err := client.RbacV1().ClusterRoleBindings().Delete(context.TODO(), "cluster-admin", metav1.DeleteOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
 			// Grant 'test-user' permission to sign CertificateSigningRequests with the specified signerName.
 			const username = "test-user"
 			grantUserPermissionToSignFor(t, client, username, test.allowedSignerName)
@@ -100,6 +105,12 @@ dgA7Fe4tMAoGCCqGSM49BAMCA0gAMEUCIQCTT1YWQZaAqfQ2oBxzOkJE2BqLFxhz
 -----END CERTIFICATE-----
 Trailing non-PEM content
 `)
+
+			// superuser should always have permission to sign; dry-run so we don't actually modify the CSR so the non-superuser can attempt as well
+			if _, err := client.CertificatesV1().CertificateSigningRequests().UpdateStatus(context.TODO(), csr, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}}); err != nil {
+				t.Errorf("expected no superuser error but got: %v", err)
+			}
+
 			_, err := testuserClient.CertificatesV1().CertificateSigningRequests().UpdateStatus(context.TODO(), csr, metav1.UpdateOptions{})
 			if err != nil && test.error != err.Error() {
 				t.Errorf("expected error %q but got: %v", test.error, err)
```

**Pull request merged**